### ISSUE TITLE
about the training process

### ISSUE THREAD

**Issue by yuanyuansiyuan** [HaojiHu/Sets2Sets#4]
It semms that when calculating the WMSE loss, we calculate the mse between a softmax probability vector generated by decoder and the groundtruth multi-hot vector.
When testing, we choose top-k to get the multi-hot prediction vector.
Have you ever tried when training, we also choose the topk from  o(vi) generated by the decoder, and then calculate the distance between two multi-hot vector?
In this way, the operations of training and testing is consistent.

Waiting for your response, thank you!

**Comment by HaojiHu** [comment]
No, I haven't tried this and I don't think this way can happen in the real application. We cannot predict which k the users will use in the test steps. We can just assume user can use any k based on their needs.

**Comment by yuanyuansiyuan** [comment]
i see, thank you for your explanation!

**Comment by yuanyuansiyuan** [comment]
it says that when doing inference, top-k strategy is just one possible greedy choice, isn't it?

**Comment by HaojiHu** [comment]
Yes. You can think about predicting the actual set size but it will further increase the complexity.
